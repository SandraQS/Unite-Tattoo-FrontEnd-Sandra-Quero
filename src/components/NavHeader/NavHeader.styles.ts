import { StyleSheet } from "react-native";
import { fontSize, colors } from "../../styles/uniteTatto.styles";

export default StyleSheet.create({
  logoContainter: {},
  nameUserContainter: {},
  nameUserText: {
    color: colors.darkBrown,
    fontSize: 15,
    paddingEnd: 10,
  },

  navContainter: {
    backgroundColor: colors.white,
    borderBottomColor: colors.darkBrown,
    borderBottomWidth: 1,
    flexDirection: "row",
    justifyContent: "space-between",
    paddingHorizontal: 20,
    paddingVertical: 10,
  },

  userContainter: {
    alignItems: "center",
    flexDirection: "row",
  },
});
