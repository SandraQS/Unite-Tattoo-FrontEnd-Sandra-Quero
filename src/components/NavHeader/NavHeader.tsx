import React from "react";
import { Text, TouchableOpacity, View } from "react-native";
import AutoHeightImage from "react-native-auto-height-image";
import { NavHeaderProps } from "../../types/components.types";

import styles from "./NavHeader.styles";
const NavHeader = ({ nameUser }: NavHeaderProps) => {
  return (
    <View style={styles.navContainter}>
      <View style={styles.logoContainter}>
        <TouchableOpacity onPress={() => {}} activeOpacity={0.6}>
          <AutoHeightImage
            width={100}
            source={require("../../assets/utoo-logo.png")}
          />
        </TouchableOpacity>
      </View>

      <View>
        <TouchableOpacity
          onPress={() => {}}
          activeOpacity={0.6}
          style={styles.userContainter}
        >
          <View style={styles.nameUserContainter}>
            <Text style={styles.nameUserText}>{nameUser}</Text>
          </View>

          <View style={styles.logoContainter}>
            <AutoHeightImage
              width={30}
              source={require("../../assets/icono-user.png")}
            />
          </View>
        </TouchableOpacity>
      </View>
    </View>
  );
};

export default NavHeader;
