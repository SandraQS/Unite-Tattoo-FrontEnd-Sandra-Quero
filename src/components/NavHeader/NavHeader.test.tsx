import React from "react";
import { fireEvent, render } from "@testing-library/react-native";
import { NavigationContainer } from "@react-navigation/native";
import configureStore from "../../redux/store";
import { Provider } from "react-redux";
import NavHeader from "./NavHeader";

describe("Given a NavHeader component", () => {
  describe("When it is rendered", () => {
    test("Then it should show match the last snapshot", () => {
      const nameUser = "Sandra";
      const store = configureStore();

      const screen = render(
        <Provider store={store}>
          <NavigationContainer>
            <NavHeader nameUser={nameUser} />
          </NavigationContainer>
        </Provider>
      );

      expect(screen).toMatchSnapshot();
    });
  });
});
