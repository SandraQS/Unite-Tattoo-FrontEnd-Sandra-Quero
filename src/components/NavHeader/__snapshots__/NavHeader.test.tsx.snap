// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a NavHeader component When it is rendered Then it should show match the last snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFF",
      "borderBottomColor": "#7C6B53",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 20,
      "paddingVertical": 10,
    }
  }
>
  <View
    style={Object {}}
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        onError={[Function]}
        source={
          Object {
            "testUri": "../../../src/assets/utoo-logo.png",
          }
        }
        style={
          Array [
            Object {
              "height": 0,
              "width": 100,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
  <View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "opacity": 1,
        }
      }
    >
      <View
        style={Object {}}
      >
        <Text
          style={
            Object {
              "color": "#7C6B53",
              "fontSize": 15,
              "paddingEnd": 10,
            }
          }
        >
          Sandra
        </Text>
      </View>
      <View
        style={Object {}}
      >
        <Image
          onError={[Function]}
          source={
            Object {
              "testUri": "../../../src/assets/icono-user.png",
            }
          }
          style={
            Array [
              Object {
                "height": 0,
                "width": 30,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
  </View>
</View>
`;
